### ISSUE TITLE
Policy Service sometimes doesn’t enable automerge

### ISSUE THREAD

**Issue by Trenly** [microsoft/winget-pkgs#142854]
Sometimes Auto-Merge isn’t being enabled and it isn’t clear why. The triggers are working for most PRs, but a few are getting into a state which require a manual merge. I expect this is an issue with the policy service itself either running into rate limits or internal errors

**Comment by Trenly** [comment]
[Policy] Issue-Bug
[Policy] Area-External
[Policy] Needs-Attention
[Policy] Area-Bots

**Comment by Trenly** [comment]
This is still in ongoing discussion with the GitOps team

**Comment by R-Adrian** [comment]
Pull https://github.com/microsoft/winget-pkgs/pull/149544

The package for Dropbox version version 197.4.7571 seems also affected by this - the commit is apparently merged but not published.

![image](https://github.com/microsoft/winget-pkgs/assets/16319504/2554c57f-84cd-42e9-9deb-5c429e74a2ab)

**Comment by Trenly** [comment]
> Pull #149544
> 
> The package for Dropbox version 197.4.7571 seems also affected by this - the commit is apparently merged but not published.

This has nothing to do with the policy service not enabling automerge

**Comment by R-Adrian** [comment]
> 
> This has nothing to do with the policy service not enabling automerge

it's been 19+ hours already since the merge happened - shouldn't it have been picked up by winget cli already?

**Comment by Trenly** [comment]
> > This has nothing to do with the policy service not enabling automerge
> 
> it's been 19+ hours already since the merge happened - shouldn't it have been picked up by winget cli already?

The publishing pipeline was paused for a deployment. The publish pipeline is completely separate from the policy service. 

You can see in the PR you linked that the Policy Service *did* enable the automerge
![image](https://github.com/microsoft/winget-pkgs/assets/12611259/a81997c5-fa83-4a24-aeee-f5f24ee3e7be)

**Comment by Trenly** [comment]
.

**Comment by Trenly** [comment]
.

**Comment by Trenly** [comment]
.

**Comment by Trenly** [comment]
For anyone watching this, it continues to be an issue

**Comment by stephengillie** [comment]
> For anyone watching this, it continues to be an issue

It might look odd, with every linked issue being merged, but this is because it occurs on the merge step of the PR lifecycle, with the highly-utilized workaround of a manual merge being performed.

**Comment by DuckDuckStudio** [comment]
Is that the same issue?
![image](https://github.com/user-attachments/assets/7ffe0b48-09ab-432f-95fb-d1340bc0b11a)
Then
![image](https://github.com/user-attachments/assets/058e3698-4f2a-4c06-a9eb-cf18d987fffa)

**Comment by Trenly** [comment]
> Is that the same issue? ![image](https://private-user-images.githubusercontent.com/89643991/350200411-7ffe0b48-09ab-432f-95fb-d1340bc0b11a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzNTkxMTcsIm5iZiI6MTcyMTM1ODgxNywicGF0aCI6Ii84OTY0Mzk5MS8zNTAyMDA0MTEtN2ZmZTBiNDgtMDlhYi00MzJmLTk1ZmItZDEzNDBiYzBiMTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE5VDAzMTMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcwZmVmMzIyMzcwZDNhM2ZkMTE2NGVkOTZjNjMzNGQ1MjQyNTg5MWQ5ZTk5YzAzMGNkMDhkNmE5ZjNmNjAwNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lPSA3v5CeIUv5-vnMUwo-7a8N7FhvYlXT3nYM8eRCIM) Then ![image](https://private-user-images.githubusercontent.com/89643991/350200444-058e3698-4f2a-4c06-a9eb-cf18d987fffa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzNTkxMTcsIm5iZiI6MTcyMTM1ODgxNywicGF0aCI6Ii84OTY0Mzk5MS8zNTAyMDA0NDQtMDU4ZTM2OTgtNGYyYS00YzA2LWE5ZWItY2YxOGQ5ODdmZmZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE5VDAzMTMzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1MDZlNjA4ODdlY2RiNDhiNDM1NjQ0ZmNlNjYzYzBmN2Q5NTNhNjJiY2VlY2JlZmViODVjZmM2ZmU4OTVkOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7hVHRGvz2BbWwMB_ExSx4HZ83C5NeJqli3x0hDWePuk)
> 
> #163740

No, it isn’t. In the screenshot it was enabled and then subsequently disabled, most likely due to many merges going into main all around the same time. This issue is tracking the fact that sometimes the policy service just doesn’t enable it at all in the first place

**Comment by stephengillie** [comment]
This issue seems to happen much less frequently than before. 
- There might be a correlation with the https://github.com/microsoft/winget-pkgs/labels/Internal-Error-Dynamic-Scan label, as when resource depletion causes this label to be applied to PRs in higher volumes, the Policy Service also seems to have more issues enabling automerge. 
- Packages with the https://github.com/microsoft/winget-pkgs/labels/Internal-Error-Dynamic-Scan label seem to have a higher rate of not having automerge enabled. I do not know if this is related to the first item, or merely a kind of "label prejudice."

**Comment by Trenly** [comment]
@stephengillie - Dragos seemed to think this issue was fully resolved; If you're still seeing issues where automerge isn’t enabled, can you link them?

For reference - Auto-Merge should be enabled when the https://github.com/microsoft/winget-pkgs/labels/Validation-Completed label is added, so no PRs that have https://github.com/microsoft/winget-pkgs/labels/Internal-Error-Dynamic-Scan should have it enabled unless they either pass validation later or the label is manually added.

**Comment by stephengillie** [comment]
- https://github.com/microsoft/winget-pkgs/pull/167167
- https://github.com/microsoft/winget-pkgs/pull/166335

**Comment by Trenly** [comment]
> * [Tencent.WeType version 1.2.3.652 #167167](https://github.com/microsoft/winget-pkgs/pull/167167)
> * [Automatic update of PLaSK to 2024.08.05 #166335](https://github.com/microsoft/winget-pkgs/pull/166335)

First one never received Validation-Completed label, and the second one it looks like it did enable and was later disabled because changes were pushed - not sure why it didn’t re-enable later though

**Comment by mdanish-kh** [comment]
I think I saw this again in

- https://github.com/microsoft/winget-pkgs/pull/191508

@Trenly  - The issue was closed by a scheduled run as it's got an Area-External label. Should we re-open the issue?

**Comment by Trenly** [comment]
@mdanish-kh - It looks like it was enabled initially, and then GitHub Core Logic disabled it due to too many PRs merging in all at the same time. Give it some time for the retry logic of wingetbot to kick in and see what happens

**Comment by Trenly** [comment]
Reopen with reason: Observed again;

* https://github.com/microsoft/winget-pkgs/pull/250228
